.github/workflows/test-node-bindings.yml

```diff
@@ -0,0 +1,48 @@
+name: Test Node Bindings
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    # only run tests when related changes are made
+    paths:
+      - ".github/workflows/test-node-bindings.yml"
+      - "bindings_node/**"
+      - "dev/**"
+      - "mls_validation_service/**"
+      - "xmtp_api_grpc/**"
+      - "xmtp_cryptography/**"
+      - "xmtp_id/**"
+      - "xmtp_mls/**"
+      - "xmtp_proto/**"
+      - "xmtp_v2/**"
+      - ".node-version"
+      - "Cargo.toml"
+      - "Cargo.lock"
+      - "rust-toolchain"
+
+jobs:
+  test:
+    runs-on: warp-ubuntu-latest-x64-16x
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: |
+            .
+            bindings_node
+
+      - name: Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Start Docker containers
+        run: dev/up
+
+      - name: Run cargo test on bindings_node
+        run: |
+          cargo test --manifest-path bindings_node/Cargo.toml -- --test-threads=2
```

Comment:
Would it make sense to reuse `test.yml` so that we just need to add the `bindings_node` to cache and then run the tests.

Reply:
this is another instance where running these tests is optimized by path changes. see `paths-ignore` above.

Reply:
Yeah CI does't run the test when the changes are only in `bindings_node` but I feel creating a new file would make the `workflows/` folder a bit messy while not saving a lot. 
Anyways, it's very nit feel free to disregard.